Source/core/DataElement.h

```diff
@@ -225,15 +225,12 @@ namespace Core {
         inline void Align()
         {
             if (m_Buffer != nullptr) {
-                size_t size(static_cast<size_t>(m_MaxSize));
-                void* newPointer = m_Buffer;
-                if (std::align(sizeof(TYPE), sizeof(TYPE), newPointer, size)) {
-                    uint8_t adjust(static_cast<uint8_t>(m_MaxSize - size));
-                    m_Buffer = reinterpret_cast<uint8_t*>(newPointer);
-                    m_Size = (adjust < m_Size ? (m_Size - adjust) : 0);
-                    m_MaxSize -= adjust;
-                    TRACE_L1("Aligning the memory buffer by %d bytes to %p !!!\n\n", adjust, m_Buffer);
-                }
+                uint8_t* newPointer = reinterpret_cast<uint8_t*>(Alignment(sizeof(TYPE), m_Buffer));
+                uint8_t adjust(static_cast<uint8_t>(newPointer - m_Buffer));
+                m_Buffer = newPointer;
+                m_Size = (adjust < m_Size ? (m_Size - adjust) : 0);
+                m_MaxSize -= adjust;
```

Comment:
Is it certain m_MaxSize >=  adjust is always true? (unsigned so will get rather large value otherwise)